lib/train/plugins/cache_connection.rb

```diff
@@ -0,0 +1,45 @@
+# encoding: utf-8
+
+class Train::Plugins::Transport
+  class CacheConnection
+    attr_accessor :cache_enabled
```

Comment:
Ruby nitpick, I'd rather see a `cache_enabled?` method that looks at this hash and returns true/false... just to keep consistency with `?` methods returning booleans. If you choose to take this advice, this will obviously require some changes in BaseConnection, but not too many.